tachyon/math/finite_fields/small_prime_field_mont.h

```diff
@@ -0,0 +1,237 @@
+// Copyright 2022 arkworks contributors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.arkworks and the LICENCE-APACHE.arkworks
+// file.
+
+#ifndef TACHYON_MATH_FINITE_FIELDS_SMALL_PRIME_FIELD_MONT_H_
+#define TACHYON_MATH_FINITE_FIELDS_SMALL_PRIME_FIELD_MONT_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <optional>
+#include <string>
+
+#include "tachyon/base/logging.h"
+#include "tachyon/base/random.h"
+#include "tachyon/base/strings/string_number_conversions.h"
+#include "tachyon/base/strings/string_util.h"
+#include "tachyon/build/build_config.h"
+#include "tachyon/math/base/egcd.h"
+#include "tachyon/math/base/gmp/gmp_util.h"
+#include "tachyon/math/finite_fields/prime_field_base.h"
+
+namespace tachyon::math {
+
+template <typename Config>
+class PrimeFieldGpu;
+
+// A prime field is finite field GF(p) where p is a prime number.
+template <typename _Config>
+class PrimeField<_Config, std::enable_if_t<!_Config::kIsSpecialPrime &&
+                                           (_Config::kModulusBits <= 32) &&
+                                           _Config::kUseMontgomery>>
+    final : public PrimeFieldBase<PrimeField<_Config>> {
+ public:
+  constexpr static size_t kModulusBits = _Config::kModulusBits;
+  constexpr static size_t kLimbNums = (kModulusBits + 63) / 64;
+  constexpr static size_t N = kLimbNums;
+
+  using Config = _Config;
+  using BigIntTy = BigInt<N>;
+  using MontgomeryTy = BigInt<N>;
```

Comment:
Are these two aliases for same type for readability? How about using `BigIntTy` for both cases?

Reply:
Yeap, I agree that the naming convention I used initially was not optimal. The `MontgomeryTy` type is intended for facilitating data transfer between various prime field implementations, such as `PrimeFieldGpu` and `PrimeField`. You can find numerous examples by searching for `ToMontgomery()`. I believe it would be beneficial to rename these to convey a clearer message